services/device-registry-file/src/main/java/org/eclipse/hono/deviceregistry/file/FileBasedDeviceBackend.java

```diff
@@ -175,9 +175,8 @@ public FileBasedDeviceBackend(
     }
 
     @Override
-    public Future<OperationResult<List<DeviceWithId>>> searchDevices(final String tenantId, final Optional<Integer> limit,
-                                                                     final Optional<Integer> offset, final Optional<List<SortingOption>> sorting, final Optional<List<Filter>> filters, final Span span) {
-        return registrationService.searchDevices(tenantId, limit, offset, sorting, filters, span);
+    public Future<OperationResult<List<DeviceWithId>>> searchDevices(final String tenantId, final int pageSize, final int pageOffset, final List<Filter> filters, final List<Sort> sortOptions, final Span span) {
```

Comment:
Could you please rebase again with the master? `searchDevices(...)` doesn't return `List<DeviceWithId>` anymore rather `SearchDevicesResult`.

Reply:
Done.